docs/rfcs/accepted/0006-hardening-against-toctou/README.md

```diff
@@ -0,0 +1,70 @@
+- Feature Name: Hardening against TOCTOU
+- Author(s): Bastien Simondi
+- Start Date: 2023-04-07
+- RFC PR: 0006
+- Leader(s): Eric Berry
+
+## Introduction
+
+Add mechanics to harden the code against Time-of-Check / Time-of-Use (TOCTOU) attacks.
+
+## Motivation / use-cases
+
+TOCTOU is a class of security vulnerabilities that is hard to avoid in complex code bases, and can 
+be especially challenging when exchanging data between different execution environment with shared 
+memories.
+All parameters and fields coming from the outside world (anything calling inside of the code) 
+should be sanitized, but special concern is to make sure there is no possibilities for the value 
+to change between the sanitization and the use of the value. Even a very short window can present 
+opportunities and be abused. In general is it preferable to cache (create a copy of) all structures 
+and parameters coming from outside of the code.
+Abuse of TOCTOU attacks can lead to serious security vulnerabilities directly or indirectly 
+leading to data compromise or arbitrary data execution in the worse cases.
+
+## Updates/Obsoletes
+
+-
```

Comment:
Recommend using "None" in all blank sections.

Reply:
Done